app/views/unattended/report_templates/ansible_-_ansible_inventory.erb

```diff
@@ -174,10 +174,13 @@ model: ReportTemplate
         inventory_data.update('host_collections': host_collections)
       end
       inventory_data.update('host_parameters': host.params) if input_hostparameters
-      inventory_data.update('ipv4': host.ip) if input_ipv4
-      inventory_data.update('ipv6': host.ip6) if input_ipv6
-      inventory_data.update('subnet': host.subnet) if input_subnet
-      inventory_data.update('subnet6': host.subnet6) if input_subnet6
+
+      net_sources = [host]
+      net_sources.unshift(host.execution_interface) if host.respond_to?(:execution_interface)
+      inventory_data.update('ipv4': net_sources.find { |s| s.ip }&.ip) if input_ipv4
+      inventory_data.update('ipv6': net_sources.find { |s| s.ip6 }&.ip6) if input_ipv6
+      inventory_data.update('subnet': net_sources.find { |s| s.subnet }&.subnet) if input_subnet
```

Comment:
Thinking out loud: can it happen that you have an ip that comes from the execution interface but the subnet that comes from the host? Wouldn't that be confusing? Same for ip6 & subnet6 of course.

Reply:
fixed